File: cpp/cpp_gui/src/Function.cpp
#include <cmath>


#include "Function.hpp"


void Function::calculate(f64 begin, f64 end, f64 step, f64& data, const std::function<f64 (f64, f64)> integral_function) {
    while (begin <= end) {
        data += integral_function(begin, step);
        begin += step;
    }
}

File: java/java_gui/src/main/java/com/example/lab_10/Function.java
package com.example.lab_10;

public class Function extends Thread implements Runnable {
        public Function() {}

    public double calculate(double begin, double end, double step, Lab10.Function2<Double, Double> integral_function) {
        double result = 0.0;

        while (begin <= end) {
            result += integral_function.apply(begin, step);
            begin += step;
        }

        return result;
    }
}
